play.rb

```diff
@@ -0,0 +1,100 @@
+$LOAD_PATH << '.'
+require 'deck'
+require 'dealer'
+require 'player'
+
+def check_bust(player, dealer)
+  if player.total > 21
+    puts 'PLAYER BUST!! Dealer wins! '\
+    "Player total: #{player.total}, Dealer total: #{dealer.total}"
+    end_game
+  end
+  if dealer.total > 21
+    puts 'DEALER BUST!! Player wins! '\
+    "Player total: #{player.total}, Dealer total: #{dealer.total}"
+    end_game
+  end
+end
+
+def check_win(player, dealer)
+  d = 21 - dealer.total # How close dealer total is to 21
+  pl = 21 - player.total # How close player total is to 21
+  puts 'TIE GAME!' if d == pl
+  puts 'DEALER WINS! '\
+  "Player total: #{player.total}, Dealer total: #{dealer.total}" if d < pl
+  puts 'PLAYER WINS! '\
+  "Player total: #{player.total}, Dealer total: #{dealer.total}" if d > pl
+end
+
+def end_game
+  puts 'Thanks for playing! Seeya next time'
+  exit!
+end
+
+# Game starts here...
+puts 'Welcome to Blackjack!'
+deck = Deck.new
+
+p1 = Player.new
+p1.hand = [deck.game_deck[0], deck.game_deck[1]]
+p1.total = p1.calculate_total
+
+dealer = Dealer.new
+dealer.hand = [deck.game_deck[2], deck.game_deck[3]]
+dealer.total = dealer.calculate_total
+
+puts 'Cards in your hand: '
+puts p1.hand.each(&:to_s)
+puts "Your total is: #{p1.total}"
+
+puts "The dealer has: \nX"
+puts dealer.hand.drop(1).each(&:to_s)
+
+user_response = ''
+
+# Keeping track of cards in the deck, 4 have already been dealt
+card_num = 4
+
+# Continue asking until player stands
+until user_response.casecmp('stand') == 0
+  puts 'Hit or stand?'
+  user_response = gets.chomp
+
+  if user_response.casecmp('hit') == 0
+    puts 'You hit!'
+    p1.hand << deck.game_deck[card_num]
```

Comment:
By having `card_num` here, you are tracking the Deck state outside of the object. Consider moving the current state of the Deck into it's class. You could add a method like `deck.draw` to get the next card and let the Deck maintain it's internal state.

```ruby
  p1.hand << deck.draw
```